testsupport/tiers/checks.go

```diff
@@ -1062,87 +1057,62 @@ func count(resource corev1.ResourceName) corev1.ResourceName {
 func numberOfToolchainRoles(number int) spaceRoleObjectsCheck {
 	return func(t *testing.T, ns *corev1.Namespace, memberAwait *wait.MemberAwaitility, owner string) {
 		roles := &rbacv1.RoleList{}
-		err := memberAwait.WaitForExpectedNumberOfResources(ns.Name, "Roles", number, func() (int, error) {
+		memberAwait.WaitForExpectedNumberOfResources(t, "Roles", number, func() (int, error) {
 			err := memberAwait.Client.List(context.TODO(), roles, providerMatchingLabels, client.InNamespace(ns.Name))
 			require.NoError(t, err)
 			return len(roles.Items), err
 		})
-		if err != nil {
-			rs := make([]string, len(roles.Items))
-			for i, r := range roles.Items {
-				rs[i] = r.Name
-			}
-			t.Logf("found %d roles: %s", len(roles.Items), spew.Sdump(rs))
-			if nsTmplSet, err := memberAwait.WaitForNSTmplSet(owner); err == nil {
```

Comment:
why printing roles and waiting for NSTmplSet is not needed anymore ?

Reply:
good point, I was a bit too aggressive with the removals. 
Let me restore that (with some refactoring)

Reply:
see 0117c25

Reply:
looks good! was `memberAwait.WaitForNSTmplSet(owner);` useless? It doesn't seem to be present in the new code changes ..

Reply:
restored in a4e60f1